Runtime/Interaction/SnapZonePreviewDrawer.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using UnityEngine;
+
+namespace Innoactive.Creator.XRInteraction
+{
+    [ExecuteInEditMode]
+    public class SnapZonePreviewDrawer : MonoBehaviour
+    {
+        public SnapZone Parent;
+        
+        private void Awake()
+        {
+            if (Application.isPlaying)
+            {
+                DestroyPreview();
+                DestroyImmediate(this);
+            }
+        }
+
+        private void Update()
+        {
+            bool isPreviewing = gameObject.GetComponent<MeshFilter>() != null;
+            
+            if (Parent.ShowHighlightInEditor && Parent.PreviewMesh && isPreviewing == false)
+            {
+                MeshFilter meshFilter =  gameObject.AddComponent<MeshFilter>();
+                meshFilter.sharedMesh = Parent.PreviewMesh;
```

Comment:
Does this work if the parent has a SkinnedMeshRenderer?

Reply:
dont know :x

Reply:
I think it should, I'll test it

Reply:
It does work